src/main/java/com/_up/megastore/services/implementations/ProductService.java

```diff
@@ -50,12 +50,21 @@ public Product findProductByIdOrThrowException(UUID productId) {
     }
 
     @Override
-    public void deleteProductById(UUID productId){
+    public void deleteProduct(UUID productId){
+        ifProductIsNotDeletedThrowException(productId);
         Product product = findProductByIdOrThrowException(productId);
         product.setDeleted(true);
         productRepository.save(product);
     }
 
+    @Override
+    public void ifProductIsNotDeletedThrowException(UUID productId){
```

Comment:
No hace falta que esto sea public ya que sólo va a ser utilizado dentro de esta misma clase. Además el nombre está mal, estás diciendo que si el producto no está eliminado tirás la excepción cuando es lo contrario
```suggestion
    private void ifProductIsAlreadyDeletedThrowException(UUID productId) {
```